### PR TITLE
Selenium: adding new cases that covered Admin UI page

### PR DESCRIPTION
**Pull request by mouse1203**
Adding new cases that covered Admin UI page:
- checking paging
- checking search field
- checking start all, stop all buttons
- checking stop, start, spawn page, access buttons
Updating test_spawn_pending_server_ready:
- adding await wait_for_ready
- moving a line await in_thread

### CODE REVIEW

File: jupyterhub/tests/selenium/test_browser.py [hunk 1]
```diff
@@ -990,3 +991,300 @@ async def test_oauth_page(
...
+    return [create_user_with_scopes(["users"]) for i in range(1, n)]
+
+
+async def test_open_admin_page(app, browser, create_user_with_scopes):
+    admin_user = create_admin_user(create_user_with_scopes)
+    await open_admin_page(app, browser, admin_user)
+    assert '/hub/admin' in browser.current_url
+
+
+def check_btns(browser, name):
```
[the remark this answers is not in the document]
Comment (mouse1203) [reply jupyterhub/tests/selenium/test_browser.py]: hi Min, thank you for the review, the function name and function itself have been fixed.

File: jupyterhub/tests/selenium/test_browser.py [hunk 2]
```diff
@@ -990,3 +991,300 @@ async def test_oauth_page(
...
+    all_btns = browser.find_elements(
+        By.XPATH, '//*[@data-testid="user-row-server-activity"]//button'
```
[the remark this answers is not in the document]
Comment (mouse1203) [reply jupyterhub/tests/selenium/test_browser.py]: fixed

File: jupyterhub/tests/selenium/test_browser.py [hunk 3]
```diff
@@ -990,3 +991,300 @@ async def test_oauth_page(
...
+    user_scopes = [
+        "admin-ui",
+        "admin:users",
+        "admin:servers",
+        "admin:groups",
+        "access:servers",
+        "list:users",
+    ]
+    admin_user = create_user_with_scopes(*user_scopes)
+    return admin_user
```
[the remark this answers is not in the document]
Comment (mouse1203) [reply jupyterhub/tests/selenium/test_browser.py]: fixed, removed this function and start to use fixture admin_user

File: jupyterhub/tests/selenium/test_browser.py [hunk 4]
```diff
@@ -990,3 +991,300 @@ async def test_oauth_page(
...
+        == len(btns["secondary"])
+        == len(users)
+        == users_count_db
+    )
+    assert not btns["stop-button"] and not btns["primary"]
+
+    # start all servers via the Start All
+    await click_all_btns(browser, "start-all", "stop-button")
+    # Start All and Stop All are still displayed
+    assert start_all_btn.is_displayed(), stop_all_btn.is_displayed()
```
[the remark this answers is not in the document]
Comment (mouse1203) [reply jupyterhub/tests/selenium/test_browser.py]: fixed, added "and"

File: jupyterhub/tests/selenium/test_browser.py [hunk 5]
```diff
@@ -990,3 +991,300 @@ async def test_oauth_page(
...
+        assert btn_previous.get_dom_attribute("class") == "active-pagination"
+        # click on Previous button
+        await click_and_wait(browser, buttons_number=1)
+    else:
+        assert btn_previous.get_dom_attribute("class") == "inactive-pagination"
+        assert btn_next.get_dom_attribute("class") == "inactive-pagination"
+
+
+async def test_search_on_admin_page(
+    app, browser, create_user_with_scopes, search_value="user", n=60
```
[the remark this answers is not in the document]
Comment (mouse1203) [reply jupyterhub/tests/selenium/test_browser.py]: fixed, added parametrizing to functions: test_search_on_admin_page, test_start_stop_server_on_admin_page.
But I added the next number [10, 47, 48, 49, 110] as I have 2 users (by default) so at the end in this case we have the same in total numbers of users as you proposed.

**Comment by minrk** [comment]
Great, thanks!

**Pull request merged**